### PR TITLE
SJQ2T-229: BE - Basic file upload/download

### PR DESCRIPTION
**Pull request by ivanfhtec**
* no tests

* no documentation

* no request validation

* no logs

seems legit

### CODE REVIEW

File: backend/src/main/java/com/htecgroup/skynest/repository/FileMetadataRepository.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import com.htecgroup.skynest.model.entity.BucketEntity;
+import com.htecgroup.skynest.model.entity.FileMetadataEntity;
+import com.htecgroup.skynest.model.entity.FolderEntity;
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.UUID;
+
+public interface FileMetadataRepository extends CrudRepository<FileMetadataEntity, UUID> {
+  boolean existsByNameAndBucketAndParentFolder(
```
Comment (DraganMitrasinovicHtec) [inline backend/src/main/java/com/htecgroup/skynest/repository/FileMetadataRepository.java:11]: This checks for a unique combination of these 3 attributes, like a composite key?

Comment (ivanfhtec) [reply backend/src/main/java/com/htecgroup/skynest/repository/FileMetadataRepository.java:11]: and it works somehow

[the db will also have triggers for the same thing](https://github.com/JDP2022SkyNest/SkyNest/pull/125)

File: backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+  private String storeFileContentsToDatabase(MultipartFile multipartFile) throws IOException {
+    Object objectId =
+        operations.store(
+            multipartFile.getInputStream(),
+            multipartFile.getOriginalFilename(),
+            multipartFile.getContentType());
+    return objectId.toString();
+  }
+
```
Comment (DraganMitrasinovicHtec) [inline backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java:94]: The logic here seems a little convoluted, lots of helper functions. Mentors would probably ask for the checks to be removed to a separate class (or a custom annotation :)), but I can live with this.

**Review by DraganMitrasinovicHtec — DISMISSED** [review]
Rough around the edges, but good for starters

File: backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java (hunk 2 above)
Comment (ivanfhtec) [reply backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java:94]: those helper functions are supposed to make this easier to read

not sure if it worked though

Comment (DarkoVeljanoskiHtec) [reply backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java:94]: I agree with this, it's kinda hard to read it like this, but maybe it's good practice like this I don't know.

Comment (DraganMitrasinovicHtec) [reply backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java:94]: Maybe it would be okay if the checks are somewhere else.

**Comment by ivanfhtec** [comment]
now has docs and logs
also checks available space in buckets and doesn't allow uploads to deleted buckets

**Review by DraganMitrasinovicHtec — APPROVED** [review]
Looks great now

**Pull request merged**